source/dofs/dof_tools_constraints.cc

```diff
@@ -3905,8 +3905,8 @@ namespace DoFTools
   {
     namespace Assembler
     {
-      struct Scratch
-      {};
+      // We don't actually need a scratch object, so use an empty class for it.
+      using Scratch = std::tuple<>;
```

Comment:
Why is that better than using a simple empty tag-like struct?

Reply:
I don't actually think it matters much to me. It was mostly an attempt at keeping the number of symbols we create small(er). It seemed silly to declare an empty class that we never use. But I'd be happy to revert that part if you prefer.

I did think on the way to work this morning about other, more obscure ways to create empty classes. This came to mind:
```
using Scratch = decltype([](){});
```
I think that has some really nice obfuscation going for it. This one uses a standard type whose name really doesn't give away at all what it is doing:
```
using Scratch = std::monostate;
```
I imagine I could come up with other ways of writing "empty type", but perhaps the ones above are enough? ;-)

Reply:
One can also iterate this if it is important to also bring in `<...>` brackets:
```
using Scratch = decltype(declval<decltype([](){})>());
```

Reply:
Are you trying to make things as complicated as possible? :-)
No harm in just doing ``using Scratch = int;``.

Reply:
Not really trying to make it complicated ;-)

Do I understand right that I should just undo the changes that pertain to the empty struct?